sdk-lib/src/androidTest/java/cash/z/ecc/android/sdk/db/DatabaseCoordinatorTest.kt

```diff
@@ -0,0 +1,179 @@
+package cash.z.ecc.android.sdk.db
+
+import androidx.test.filters.SmallTest
+import cash.z.ecc.android.sdk.test.getAppContext
+import cash.z.ecc.android.sdk.type.ZcashNetwork
+import cash.z.ecc.fixture.DatabaseNameFixture
+import cash.z.ecc.fixture.DatabasePathFixture
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+
+class DatabaseCoordinatorTest {
+
+    private val dbCoordinator = DatabaseCoordinator.getInstance(getAppContext())
+
+    @Before
+    fun clear_test_files() {
+        val databaseDir = DatabasePathFixture.new(baseFolderPath = DatabasePathFixture.DATABASE_DIR_PATH)
+        val noBackupDir = DatabasePathFixture.new(baseFolderPath = DatabasePathFixture.NO_BACKUP_DIR_PATH)
+        File(databaseDir).deleteRecursively()
+        File(noBackupDir).deleteRecursively()
+    }
+
+    // Sanity check of the database coordinator instance and its thread-safe implementation. Our aim
+    // here is to run two jobs in parallel at the same time to test mutex implementation and correct
+    // DatabaseCoordinator function call result.
+    @Test
+    @SmallTest
+    fun mutex_test() = runTest {
+        var testResult: File? = null
+
+        launch {
+            delay(1000)
+            testResult = dbCoordinator.cacheDbFile(
+                DatabaseNameFixture.TEST_DB_NETWORK,
+                DatabaseNameFixture.TEST_DB_ALIAS
+            )
+        }
+        val job2 = launch {
+            delay(1000)
+            testResult = dbCoordinator.cacheDbFile(
+                ZcashNetwork.Mainnet,
+                "TestZcashSdk"
+            )
+        }
+
+        advanceTimeBy(1001)
+
+        job2.join().also {
+            assertTrue(testResult != null)
+            assertTrue(testResult!!.absolutePath.isNotEmpty())
+            assertTrue(testResult!!.absolutePath.contains(ZcashNetwork.Mainnet.networkName))
+            assertTrue(testResult!!.absolutePath.contains("TestZcashSdk"))
+        }
+    }
+
+    @Test
+    @SmallTest
+    fun database_cache_file_creation_test() = runTest {
+        val expectedFilePath =
+            "${DatabasePathFixture.new()}/${DatabaseNameFixture.newDb(name = DatabaseCoordinator.DB_CACHE_NAME)}"
```

Comment:
When writing JVM code (and not multiplatform code), I have a preference for doing `File(directory, file).path` so that the File class takes care of generating the path separators correctly for us because they can be different across platforms.  Java usually does the right thing, and Android is Linux-only, but it is a nice habit to get into.

So something to potentially consider.

Reply:
Yes, definitely. Thanks for pointing it out!